src/wire/Socket.ts

```diff
@@ -249,10 +249,14 @@ export class InteractiveSocket extends EventEmitter {
             return;
         }
 
-        this.state = SocketState.Closing;
-        this.socket.close(1000, 'Closed normally.');
-        this.queue.forEach(packet => packet.cancel());
-        this.queue.clear();
+        if (this.state !== SocketState.Idle) {
+            this.state = SocketState.Closing;
+            this.socket.close(1000, 'Closed normally.');
+            this.queue.forEach(packet => packet.cancel());
+            this.queue.clear();
+        } else {
+            this.state = SocketState.Closing;
```

Comment:
Websocket close is a three-way handshake, this starts it off but it doesn't complete until the close event is emitted.